### ISSUE TITLE
API Gateway v2 Integration doesn't work with MOCK Integration Type

### ISSUE THREAD

**Issue by andy-mccombs-rft** [hashicorp/terraform-provider-aws#17289]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

API Gateway v2 Integration doesn't work with MOCK Integration Type as indicated in Terraform's documentation. I'm assuming that this isn't implemented yet. **That's why I opened a Feature Request and not an issue.**

### New or Affected Resource(s)

**AWS API Gateway v2**

* aws_apigatewayv2_integration

### Terraform Configuration

AWS Provider: 3.25.0

```hcl
resource "aws_apigatewayv2_integration" "example" {
  api_id           = aws_apigatewayv2_api.apigatewayv2_api.id
  integration_type = "MOCK"
}
```

**Error**
`Error: error creating API Gateway v2 integration: BadRequestException: Currently, an API with a protocol type of HTTP may only be associated with proxy integrations (AWS_PROXY, HTTP_PROXY)`

### References

Same code found [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/apigatewayv2_integration)

**Comment by ewbankkit** [comment]
@andy-mccombs-rft Thanks for raising this issue.
Indeed this is a limitation of the underlying [AWS API](https://docs.aws.amazon.com/apigatewayv2/latest/api-reference/apis-apiid-integrations-integrationid.html#apis-apiid-integrations-integrationid-model-integration).
I will update the documentation to reflect this.

**Comment by andy-mccombs-rft** [comment]
Thanks for sending the AWS API doc. I totally missed that. Thanks for the help.

**Comment by andy-mccombs-rft** [comment]
This was not resolved with version 3.27.0 of the Terraform AWS provider. I still get the error message

`Error: error creating API Gateway v2 integration: BadRequestException: Currently, an API with a protocol type of HTTP may only be associated with proxy integrations (AWS_PROXY, HTTP_PROXY)`

**Comment by ewbankkit** [comment]
@andy-mccombs-rft The underlying AWS API does not the `MOCK` value for HTTP APIs. The documentation was updated to reflect this.
